test/node/conversation-settings.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const assert = require('assert');
+const Circuit = require('../../circuit-node');
+const config = require('./config.json');
+const helper = require('./helper');
+Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
+
+let client;
+let user;
+let settings;
+describe('Settings', () => {
+    before(async () => {
+        client = new Circuit.Client(config.bot1);
+        user = await client.logon();
+    });
+
+    after(async () => {
+        await client.logout();
+    });
+
+    it('should get user settings', async () => {
+        settings = await client.getUserSettings();
+        assert(settings);
+    });
+
+    it('should set user settings and raise userSettingsChanged event', async () => {    
+        const shareLocation = settings && !settings.shareLocation || false;
```

Comment:
settings must be true, or the above test fails, so no need to check that.
Why do you add "|| false"?

Reply:
I added || false because the first time I called getUserSettings it didn't return anything, so if settings.shareLocation is undefined for some reason it will set the value to false. I don't know if that is a good way to handle it but that seemed to handle every case.

Reply:
Ok, so use:

const shareLocation = !!settings.shareLocation;

Reply:
done